torch/_inductor/ir.py

```diff
@@ -3589,6 +3594,55 @@ def apply_constraint(self):
         raise NotImplementedError
 
 
+class UserDefinedTritonKernel(ExternKernel):
+    def codegen(self, wrapper):
+        from torch._higher_order_ops.triton_kernel_wrap import kernel_side_table
+
+        kernel = kernel_side_table.get_kernel(self.kernel_idx)
```

Comment:
hmm... maybe a little late, but @Chillee I don't think that inductor's fx graph caching `TORCHINDUCTOR_FX_GRAPH_CACHE=1` will play well with this kernel_side_table.

What do you think - should we just make sure not to add to the cache if our graph uses any user triton kernels, to unblock, or fully revisit the side table?

It seems to me like the FX graph that we get (both from dynamo and AOTAutograd) will burn in a random index that's supposed to map to a user triton kernel that dynamo saw:
```
def forward(self, x_1, output_1):
    triton_kernel_wrapper_functional_proxy = torch._higher_order_ops.triton_kernel_wrap.triton_kernel_wrapper_functional(kernel_idx = 0, grid = (5,), kwargs = {'in_ptr0': x_1, 'out_ptr': output_1, 'n_elements': 5, 'BLOCK_SIZE': 16});  x_1 = #
    getitem = triton_kernel_wrapper_functional_proxy['in_ptr0']
    getitem_1 = triton_kernel_wrapper_functional_proxy['out_ptr']
    getitem_2 = triton_kernel_wrapper_functional_proxy['n_elements']
    getitem_3 = triton_kernel_wrapper_functional_proxy['BLOCK_SIZE'];  triton_kernel_wrapper_functional_proxy = None
    return getitem_1
```

But if we were to map this graph to a cached, compiled inductor graph, we have no guarantee that that index will map to the same triton kernel.

Maybe we can have the cache key also include the kernel_side_table's mapping, from index to triton kernel (or a hash of its source code)?

Reply:
Yeah this is why I was suggesting that it might be better to just include the Triton source code in the operator definition.

Reply:
We would not only need to store the source code of one function but also every single dependancy. Another downside of that is the FX graph would be impossible to read.

If this is the preferred solution, I'm happy to change it but IMO caching kernel_side_table's mapping seems like a cleaner solution.